assets/javascripts/needleeditor.js

```diff
@@ -502,11 +502,23 @@ function saveNeedle(overwrite) {
 
   fetchWithCSRF(form.action, {method: 'POST', body: new FormData(form)})
     .then(response => {
-      if (!response.ok) throw `Server returned ${response.status}: ${response.statusText}`;
-      return response.json();
+      return response
+        .json()
+        .then(json => {
+          // Attach the parsed JSON to the response object for further use
+          return {response, json};
+        })
+        .catch(() => {
+          // If parsing fails, handle it as a non-JSON response
+          throw `Server returned ${response.status}: ${response.statusText}`;
+        });
+    })
+    .then(({response, json}) => {
+      if (!response.ok) `Server returned ${response.status}: ${response.statusText}\n${json.error}`;
```

Comment:
This line doesn't seem to do much. Is this supposed to throw the "Server returned …" string?